### PR TITLE
Fix keys in sendstring_fr (resolves #9610)

### LINKED ISSUE #9610: [Bug] sendstring_french.h 9 key
## Describe the Bug

When using sendstring_french.h, sending a string with a 9 will result in a 0 sent.

## System Information

 - Keyboard:  Planck
   - Revision (if applicable):  rev6
 - Operating system: Arch linux
 - AVR GCC version: 10.1.0

 - ARM GCC version: 10.1.0

 - QMK Firmware version: 0.8.182

 - Any keyboard related software installed? 
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

## Additional Context

This bug comes from quantum/keymap_extras/sendstring_french.h:83 where the key for 0 is `FR_AGRV` and should be `FR_CCED`.

### PR DESCRIPTION
**Pull request by CorentinCAM**
## Description

This pull request fixes the issue described in #9610 (8 and 9 bad codes when sending strings in french)

## Types of Changes

- [ ] Core
- [X] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #9610

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
Good catch!

I was just about to PR this myself - there are a couple other things wrong with it. The entry for `"` should be `FR_DQUO`, and in `keymap_french.h`, `FR_CIRC` should be changed to `FR_DCIR`, and `#define FR_CIRC ALGR(FR_CCED)` added between `FR_BSLS` and `FR_AT`.

**Comment by CorentinCAM** [comment]
Fixed it!

**Comment by CorentinCAM** [comment]
Oh sorry I didn't notice your commit!

File: quantum/keymap_extras/keymap_french.h [hunk 1]
```diff
@@ -58,7 +58,7 @@
 #define FR_I    KC_I    // I
 #define FR_O    KC_O    // O
 #define FR_P    KC_P    // P
-#define FR_CIRC KC_LBRC // ^ (dead)
+#define FR_CIRC KC_DCIR // ^ (dead)
```
Comment (GreatWizard) [inline quantum/keymap_extras/keymap_french.h:61]: This change is wrong, this is really the `KC_LBRC` keycode.

File: quantum/keymap_extras/keymap_french.h [hunk 2]
```diff
@@ -146,6 +146,7 @@
 #define FR_PIPE ALGR(FR_MINS) // |
 #define FR_GRV  ALGR(FR_EGRV) // ` (dead)
 #define FR_BSLS ALGR(FR_UNDS) // (backslash)
+#define FR_CIRC ALGR(FR_CCED) // ^
```
Comment (GreatWizard) [inline quantum/keymap_extras/keymap_french.h:149]: We should give him another name, FR_CIRC is already used in many places and devs except to be the `dead ^`.

File: quantum/keymap_extras/keymap_french.h (hunk 1 above)
Comment (fauxpark) [inline quantum/keymap_extras/keymap_french.h:61]:
```suggestion
#define FR_DCIR KC_LBRC // ^ (dead)
```

File: quantum/keymap_extras/keymap_french.h (hunk 2 above)
Comment (fauxpark) [reply quantum/keymap_extras/keymap_french.h:149]: `FR_CIRC` would be the dead `^`, unless there is a non-dead equivalent present in the layout (as there is here), in which case the dead key is `FR_DCIR`.

Comment (GreatWizard) [reply quantum/keymap_extras/keymap_french.h:149]: We can make the PR breaking changes with this changes:

```
#define FR_DCIR KC_LBRC
#define FR_CIRC ALGR(FR_CCED)
```

If we do this changes, should this PR migrate keymaps in the QMK repo?

Comment (fauxpark) [reply quantum/keymap_extras/keymap_french.h:149]: I can only find one instance in the repo that isn't a default keymap and isn't using the macOS version, `layouts/community/ergodox/french_hacker/keymap.c`. If we get signoff from the contributor of that layout, it doesn't have to go through breaking changes.

**Pull request closed without merging**